### PR TITLE
EN-614: Docs consolidatior enhancement

### PR DESCRIPTION
**Pull request by 98dfsj**
## Overview

Gives the docs consolidator function the ability to traverse a tree.

## Reason

We can now nest component documents into categories. 

## Developer notes

- No additional markup or folder structure needs to be added to the component folders of the frameworks. The consolidator now just looks in `/packages/documentation/library/pages/components` and any subfolders that it finds in there and injects any content it finds from matching framework components.
- This PR also includes updates from @m7kvqbe1 to fix the issue with recursion in the sidebar which is related to the document listing

### CODE REVIEW

**Comment by m7kvqbe1** [comment]
Could we make the comments consistent and in JSDoc format?

I'd be a little bit worried about touching this script from a maintainer point of view because there aren't any tests - it's doing a lot of very nuanced stuff.

**Comment by 98dfsj** [comment]
Fair point. I'll update them now.

**Comment by m7kvqbe1** [comment]
Were you running Prettier on this for formatting?

File: packages/docs-site/docs-consolidator.js [hunk 1]
```diff
@@ -122,21 +122,39 @@ const packageLoop = () => {
...
+}
+
+// Proccesses all files found in the folderloop and if they are a markdown file they proccess them, 
+// if they are a directory then run folderloop again inside that directory.
+const matchDocs = (docsPath, doc, prefix) => {
+  const thisFile = join(docsPath, doc)
+  if (doc.match(/.md/gi)) {
+    injectInFile(thisFile, currentComponent(thisFile))
+    logOutputBuilder(prefix, doc)
+  } else {
```
Comment (m7kvqbe1) [inline packages/docs-site/docs-consolidator.js:182]: Could we get rid of the extra nesting here? Prettier might sort this out by itself.

File: packages/docs-site/docs-consolidator.js [hunk 2]
```diff
@@ -25,20 +25,30 @@ const libraryDocsFolder = resolve(__dirname, './src/library')
 const packageDocsFolder = 'src/generated-library/pages/components/'
 
 // Ensure that the package docs folder is a freshly generated copy
-rimraf.sync(packageDocsFolder)
-fs.mkdirSync(packageDocsFolder, { recursive: true })
+rimraf.sync(packageDocsFolder) // deletes old folder
+fs.mkdirSync(packageDocsFolder, { recursive: true }) // Creates new tree structure
 
-// Convert a string to kebab-case
-const kebab = text => text
```
Comment (antictor) [inline packages/docs-site/docs-consolidator.js:32]: lodash has a function for this

File: packages/react-component-library/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+node_modules/
```
Comment (antictor) [inline packages/react-component-library/.gitignore:1]: This should be in the top level, I had to do it for dist and build ealier.

File: packages/docs-site/docs-consolidator.js [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 /**
  * Consolidate component docs
- * This script finds all README.md contained within an 
+ * This script finds all README.md contained within an
  * `src/components` folder within the specified packages
  * and consolidates the contents into one file within the gatsby library.
  **/
```
Comment (m7kvqbe1) [inline packages/docs-site/docs-consolidator.js]:
```suggestion
 */
```

File: packages/docs-site/docs-consolidator.js (hunk 2 above)
Comment (98dfsj) [reply packages/docs-site/docs-consolidator.js:32]: Nice, didn't know that. Replaced it.

File: packages/react-component-library/.gitignore (hunk 3 above)
Comment (98dfsj) [reply packages/react-component-library/.gitignore:1]: It didn't work. When I merged your branch in it tried to commit those folders, hence me adding the gitignore.

**Review by m7kvqbe1 — APPROVED** [review]

**Pull request merged**